src/SqlPersistence/ScriptLocation.cs

```diff
@@ -20,20 +19,15 @@ static string GetCurrentDirectory(ReadOnlySettings settings)
         {
             return scriptDirectory;
         }
-        var entryAssembly = Assembly.GetEntryAssembly();
-        if (entryAssembly == null)
+
+        var baseDir = AppDomain.CurrentDomain.BaseDirectory;
+        var scriptDir = Path.Combine(baseDir, ScriptFolder);
+        //if the app domain base dir contains the scripts folder, return it. Otherwise add "bin" to the base dir so that web apps work correctly
+        if (Directory.Exists(scriptDir))
         {
-            var baseDir = AppDomain.CurrentDomain.BaseDirectory;
-            var scriptDir = Path.Combine(baseDir, ScriptFolder);
-            //if the app domain base dir contains the scripts folder, return it. Otherwise add "bin" to the base dir so that web apps work correctly
-            if (Directory.Exists(scriptDir))
-            {
-                return baseDir;
-            }
-            return Path.Combine(baseDir, "bin");
+            return baseDir;
         }
-        var codeBase = entryAssembly.CodeBase;
-        return Directory.GetParent(new Uri(codeBase).LocalPath).FullName;
+        return Path.Combine(baseDir, "bin");
```

Comment:
Appending `bin` to the path without checking it exists and contains the scripts folder is assuming that the only reason that the `if` failed is because we're running inside ASP.NET. I'm wondering if this should be more defensive.